### PR TITLE
Monitor for testnet and mainnet (golem-monitor #35)

### PR DESCRIPTION
**Pull request by RDProjekt**
Receiving and storing net (mainnet/testnet) value in the database.

### CODE REVIEW

File: tests/golem/monitor/test_monitor.py [hunk 1]
```diff
@@ -62,20 +73,37 @@ def test_protocol_versions(self):
         """Test whether correct protocol versions were sent."""
 
         def check(f, msg_type):
-            with mock.patch('golem.monitor.monitor.SenderThread.send') \
+            with mock.patch(
+                    'golem.monitor.transport.httptransport.'
+                    + 'DefaultHttpSender.post_json') \
                     as mock_send:
                 f()
+                time.sleep(0.005)
```
Comment (Wiezzel) [inline tests/golem/monitor/test_monitor.py]: This is not a good way to test multi-threaded code. If you want to test `SenderThread` as well add another test case.

File: tests/golem/monitor/test_monitor.py [hunk 2]
```diff
@@ -62,20 +73,37 @@ def test_protocol_versions(self):
...
+                            'cliid': 'cliid',
+                            'sessid': 'sessid',
+                            'os': 'os',
+                            'version': 'ver',
+                            'settings': mock.ANY,
+                        },
+                        'cliid': 'cliid',
+                        'sessid': 'sessid',
+                        'timestamp': mock.ANY,
+                    }
```
Comment (Wiezzel) [inline tests/golem/monitor/test_monitor.py:108]: Test should probably be renamed if you check all these data. Now it's called `test_protocol_versions` and the docstring says `"""Test whether correct protocol versions were sent."""`.

**Review by Wiezzel — CHANGES_REQUESTED** [review]
Calling `time.sleep()` in tests must be avoided.

File: tests/golem/monitor/test_monitor.py (hunk 1 above)
Comment (RDProjekt) [reply tests/golem/monitor/test_monitor.py]: We added functions for such test in another commit, we'll create pull request for it shortly

File: golem/monitor/test_helper.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import sys
-import unittest
 from uuid import uuid4
+from unittest import mock, TestCase
```
Comment (jiivan) [inline golem/monitor/test_helper.py]: Import ordering

File: tests/golem/monitor/test_monitor.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 import time
 from unittest import mock, TestCase
 from urllib.parse import urljoin
+import json
```
Comment (jiivan) [inline tests/golem/monitor/test_monitor.py]: Import ordering

File: tests/golem/monitor/test_monitor.py [hunk 5]
```diff
@@ -25,18 +26,23 @@ class TestSystemMonitor(TestCase, testutils.PEP8MixIn):
...
+        client_mock.session_id = 'sessid'
+        client_mock.config_desc = ClientConfigDescriptor()
+        client_mock.mainnet = False
         meta_data = NodeMetadataModel(
-            'cliid', 'sessid', 'os', 'ver', ClientConfigDescriptor())
+            client_mock, 'os', 'ver')
         config = MONITOR_CONFIG.copy()
         config['HOST'] = 'http://localhost/88881'
         self.monitor = SystemMonitor(meta_data, config)
+        self.monitor.start()
```
Comment (jiivan) [inline tests/golem/monitor/test_monitor.py]: Where is `.shut_down()` called?

Comment (Wiezzel) [reply tests/golem/monitor/test_monitor.py]: In `test_monitor_messages`. :stuck_out_tongue:

Comment (Wiezzel) [reply tests/golem/monitor/test_monitor.py]: But yes, it should be called in `tearDown`.

Comment (jiivan) [reply tests/golem/monitor/test_monitor.py]: If `.start()` is moved to `setUp()` then `.shutd_down()` should be also moved to`tearDown()`

File: tests/golem/monitor/test_monitor.py (hunk 2 above)
Comment (RDProjekt) [reply tests/golem/monitor/test_monitor.py:108]: Done

File: golem/monitor/test_helper.py (hunk 3 above)
Comment (RDProjekt) [reply golem/monitor/test_helper.py]: Done

File: tests/golem/monitor/test_monitor.py (hunk 4 above)
Comment (RDProjekt) [reply tests/golem/monitor/test_monitor.py]: Done

File: tests/golem/monitor/test_monitor.py (hunk 5 above)
Comment (RDProjekt) [reply tests/golem/monitor/test_monitor.py]: Done

**Review by Wiezzel — APPROVED** [review]
OK, but please remove that sleep later.

**Pull request merged**